signed/verifiers.go

```diff
@@ -47,6 +47,9 @@ func RegisterVerifier(algorithm data.SigAlgorithm, v Verifier) {
 type Ed25519Verifier struct{}
 
 func (v Ed25519Verifier) Verify(key data.PublicKey, sig []byte, msg []byte) error {
+	if key.Algorithm() != data.ED25519Key {
```

Comment:
we should also check that the result of `copy(sigBytes[:], sig)` has the same size as `ed25519.PublicKeySize`: we didn't copy less bytes than we expected.